### ISSUE TITLE
Programs/Encounter: Headers

### ISSUE THREAD

**Issue by richardmoizeau** [openmsupply/open-msupply#676]
In HPDB, encounters have headers. 

![Screen Shot 2022-09-30 at 3 48 56 PM](https://user-images.githubusercontent.com/74992958/193179565-b27fbc8c-b0ae-4471-b733-88f1dc06bbb5.png)

Suggestions: 

1. Add patient data in the encounter headers (Program ID, Patient names, DOB, Enrolment ID)
2. Add encounter data in the details panel - these data can be added through the New Encounter Modal, only the ones that would be universal for all encounters: Visit Date, Seen by [Staff Name, Position]. For the others: 
- First Visit Date: automatically calculated, date of the 1st visit of the patient
- Visit Type: if not the `Initial` Visit then it's `Follow up`

- Type of visit: maybe that the tricky one cause I'm not sure this would be universal. Three values: Standard (Visit to this facility), Refill (Visit to a refill facility supervised by this facility), Outreach (Visit to an outreach facility where outreach staff from this facility are providing services). If you choose Refill or Outreach, then you need to select a Visit Location from a dropdown list. Maybe this could be another Encounter section using a json form ? 

![Navigate to previous encounters](https://user-images.githubusercontent.com/74992958/193179728-0a21223e-542e-4353-a082-14c2161c8d84.png)
![Navigate to previous encounters (1)](https://user-images.githubusercontent.com/74992958/193179727-b4a51ef3-28c9-43b5-a5c1-6a0c94bd0b28.png)
![Navigate to previous encounters (2)](https://user-images.githubusercontent.com/74992958/193179723-f0ed0b16-2659-4540-82d0-bfc18a53f0bb.png)

**Comment by clemens-msupply** [comment]
Visit type and first visit are slightly redundant if there is a list of encounters...

The "type of visit seem to be customer specific -> not part of the header?

**Comment by richardmoizeau** [comment]
> Visit type and first visit are slightly redundant if there is a list of encounters...
> 
> The "type of visit seem to be customer specific -> not part of the header?

I agree. Let's only keep the type of visit (Outreach, Refill and Standard). If Outreach or Refill, need to select the facility in a dropdown under "Visit Location"

**Comment by richardmoizeau** [comment]
What about the details panel ?

**Comment by richardmoizeau** [comment]
Just realised that I forget that "Seen by" (or "Practitioner") should be added as one of the field in the "details panel" and should be a drop down.

**Comment by clemens-msupply** [comment]
And the "Seen by" practitioner defaults to the current user? sounds good. Do we already have a way to mark users as "Practitioners"?

**Comment by richardmoizeau** [comment]
Ideally it should but based on Poruan's explanations, I think it is going to be drop down. From what we understood from our meetings is that Practitioner are not the ones entering the data. Data are captured afterwards by data clerks working for the clinic.

**Comment by clemens-msupply** [comment]
I have done the BE and the FE api hooks for:
https://github.com/openmsupply/open-msupply/issues/926

not sure if it is easier to implemented it as part of this issue...

**Comment by CarlosNZ** [comment]
> * Add patient data in the encounter headers (Program ID, Patient names, DOB, Enrolment ID)
> 
> * Add encounter data in the details panel - these data can be added through the New Encounter Modal, only the ones that would be universal for all encounters: Visit Date, Seen by [Staff Name, Position].

Reading through this issue, I'm not totally clear which of this two options we want to go with. @richardmoizeau @clemens-msupply ?

**Comment by clemens-msupply** [comment]
The last screenshot shows an example of the header and the details panel...

**Comment by CarlosNZ** [comment]
I'll break this out into 2-3 separate issues. Stand by...

**Comment by CarlosNZ** [comment]
Remaining elements for this: #1158

**Comment by CarlosNZ** [comment]
Closing now after #1159 merged -- not fully complete, but remaining issues are collated in #1158
